src/Button/RedliningButton/RedliningButton.jsx

```diff
@@ -0,0 +1,672 @@
+        }
+        map.un('pointermove', this.onPointerMove);
+      }
+    }
+  }
+
+  /**
+   * Creates redlining vector layer and adds this to the map.
+   *
+   * @method
+   */
+  createRedliningLayer = () => {
+    const {
+      redliningLayerName,
+      map
+    } = this.props;
+
+    let redliningLayer = MapUtil.getLayerByName(map, redliningLayerName);
+
+    if (!redliningLayer) {
+      redliningLayer = new OlLayerVector({
+        name: redliningLayerName,
+        source: new OlSourceVector({
+          features: new OlCollection(),
+        }),
+        style: this.getRedliningStyleFunction
+      });
+      map.addLayer(redliningLayer);
+    }
+    this.setState({redliningLayer});
+  }
+
+  /**
+   * The styling function for the redlining vector layer, which considers
+   * different geometry types of drawn features.
+   *
+   * @param {OlFeature} feature The feature which is being styled.
+   * @return {OlStyleStyle} The style to use.
+   *
+   * @method
+   */
+  getRedliningStyleFunction = feature => {
+
+    const {
+      fillColor,
+      selectFillColor,
+      strokeColor,
+      selectStrokeColor
+    } = this.props;
+
+    const {
+      interaction,
+    } = this.state;
+
+    let useSelectStyle = false;
+    if (interaction && interaction.length > 1 && !feature.get('isLabel')) {
+      useSelectStyle = true;
+    }
+
+    switch (feature.getGeometry().getType()) {
+      case 'Point': {
+        if (!feature.get('isLabel')) {
+          return new OlStyleStyle({
+            image: new OlStyleCircle({
+              radius: 7,
+              fill: new OlStyleFill({
+                color: !useSelectStyle ? fillColor : selectFillColor
+              }),
+              stroke: new OlStyleStroke({
+                color: !useSelectStyle ? strokeColor : selectStrokeColor
+              })
+            })
+          });
+        } else {
+          return new OlStyleStyle({
+            text: new OlStyleText({
+              text: '',
+              offsetX: 5,
+              offsetY: 5,
+              font: '12px sans-serif',
+              fill: new OlStyleFill({
+                color: !useSelectStyle ? fillColor : selectFillColor
+              }),
+              stroke: new OlStyleStroke({
+                color: !useSelectStyle ? strokeColor : selectStrokeColor
+              })
+            })
+          });
+        }
+      }
+      case 'LineString': {
+        return new OlStyleStyle({
+          stroke: new OlStyleStroke({
+            color: !useSelectStyle ? strokeColor : selectStrokeColor,
+            width: 2
+          })
+        });
+      }
+      case 'Polygon':
+      case 'Circle': {
+        return new OlStyleStyle({
+          fill: new OlStyleFill({
+            color: !useSelectStyle ? fillColor : selectFillColor
+          }),
+          stroke: new OlStyleStroke({
+            color: !useSelectStyle ? strokeColor : selectStrokeColor,
+            width: 2
```

Comment:
This and some other style properties aren't configurable yet. Do you have any attempt to make them configurable, e.g. by passing an `ol.Style` for each draw type as prop?

Reply:
yeah, please make these cinfigurable

Reply:
I've opened an issue -> https://github.com/terrestris/react-geo/issues/380

Will be done in one of follow-ups :innocent: